src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,59 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Contact {
+
+    public static String EXAMPLE = "";
```

Comment:
Do you need to declare these 3 attributes?

Reply:
need, because email, phone, address inherit these 3 attributes
but i still put those 3 attributes in email, phone, address so that i can change the values at there without skimming through the whole codes (even though the codes are short XD)

Reply:
Each of the 3 objects, `Email` `Phone` and `Address` have different values for the above three variables, it is better to handle things unique to child in the child class.